### PR TITLE
Lo business insight v1

### PR DESCRIPTION
**Pull request by lao9**

### CODE REVIEW

File: app/controllers/api/v1/merchants/merchant_pending_customers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Api::V1::Merchants::MerchantPendingCustomersController < ApplicationController
+
+  def index
+    @merchant = Merchant.find(params[:merchant_id])
+    @customers = @merchant.pending_customers
+    binding.pry
```
Comment (case-eee) [inline app/controllers/api/v1/merchants/merchant_pending_customers_controller.rb:6]: don't forget to remove this pry!

File: app/models/customer.rb [hunk 2]
```diff
@@ -9,4 +9,8 @@ def self.random
       id = Customer.pluck(:id).sample
       Customer.find(id)
     end
+
+    def transactions
```
Comment (case-eee) [inline app/models/customer.rb:13]: when you write instance methods on a class, be sure you test them - and by test them i mean test them on the unit level to ensure they're working properly 👍

File: app/models/merchant.rb [hunk 3]
```diff
@@ -9,4 +9,21 @@ def self.random
...
+  def items
+    Item.where(merchant_id: id)
+  end
+
+  def pending_customers
+    Customer.joins(invoices: :transactions).where('invoices.merchant_id = ?', id).pluck('transactions.result')
+    binding.pry
+  end
+
+  def favorite_customer
```
Comment (case-eee) [inline app/models/merchant.rb:21]: same here - this logic looks solid, but make sure you test this with a unit test!

File: app/models/merchant.rb [hunk 4]
```diff
@@ -9,4 +9,21 @@ def self.random
     Merchant.find(id)
   end
 
+  def items
+    Item.where(merchant_id: id)
+  end
+
+  def pending_customers
+    Customer.joins(invoices: :transactions).where('invoices.merchant_id = ?', id).pluck('transactions.result')
+    binding.pry
```
Comment (case-eee) [inline app/models/merchant.rb:18]: make sure you remove this 👍

File: app/models/transaction.rb [hunk 5]
```diff
@@ -1,4 +1,7 @@
 class Transaction < ApplicationRecord
+  scope :success, -> { where(status: "success") }
+  scope :failed, -> { where(status: "failed") }
```
Comment (case-eee) [inline app/models/transaction.rb:3]: woooo scopes 🎉

**Review by case-eee — COMMENTED** [review]
@lao9 @josh-works i know this has already been merged, but i left you a few comments to come back to so that you can improve your eval scores. let me know if you have questions. good job 🎉

File: app/controllers/api/v1/merchants/merchant_pending_customers_controller.rb (hunk 1 above)
Comment (josh-works) [reply app/controllers/api/v1/merchants/merchant_pending_customers_controller.rb:6]: done!

File: app/models/customer.rb (hunk 2 above)
Comment (josh-works) [reply app/models/customer.rb:13]: added to waffleboard. Thanks!

File: app/models/merchant.rb (hunk 4 above)
Comment (josh-works) [reply app/models/merchant.rb:18]: 👍

File: app/models/merchant.rb (hunk 3 above)
Comment (josh-works) [reply app/models/merchant.rb:21]: all in waffleboard

File: app/models/customer.rb (hunk 2 above)
Comment (case-eee) [reply app/models/customer.rb:13]: awesome 😄

**Pull request merged**